### ISSUE TITLE
Unnecessary strict rule a bit too aggressive

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#415]
The unnecessary-strict rule is a bit overly aggressive and in fact flags itself with a warning:

``` js
"use strict";

//------------------------------------------------------------------------------
// Rule Definition
//------------------------------------------------------------------------------

module.exports = function(context) {

    return {

        "Literal": function(node) {
            if (node.value === "use strict") {
                var scope = context.getScope();

                if (scope.upper && scope.upper.isStrict) {
                    context.report(node, "Unnecessary 'use strict'.");
                }
            }
        }

    };

};
```

Result:

```
lib/rules/unnecessary-strict.js: line 17, col 31, Error - Unnecessary 'use strict'.
```

We need to be sure that developers can use the string "use strict" when it's not a pragma.
